### ISSUE TITLE
Bug: Switch Cloud Build from Container Registry to Artifact Registry

### ISSUE THREAD

**Issue by cceckman** [recursecenter/pairing-bot#52]
Per @stillgreenmoss : [Google is shutting down](https://killedbygoogle.com/) Container Registry, with Artifact Registry as the replacement. Looks like [this page](https://cloud.google.com/artifact-registry/docs/transition/transition-from-gcr) has the transition documentation.

I don't think Pairing Bot uses GCR _directly_,[^1] but the builds going through Cloud Build may? (They refer to `gcr.io/cloud-builders`, but [that's still correct](https://cloud.google.com/build/docs/cloud-builders).)

Once #51 is done and we have some builds running, it would be good to [check the usage](https://cloud.google.com/artifact-registry/docs/transition/check-gcr-usage) and see if we actually need to do something here -- or if this is just, "You have the API enabled, so we're sending you the email."

[^1]: GAE Standard Environment uses GCS to store data, but they aren't container images _per se_ -- not using `gcr.io`.

**Comment by cceckman** [comment]
Both projects report `usage: ACTIVE` from the [check tool](https://cloud.google.com/artifact-registry/docs/transition/check-gcr-usage).

Poking at the container registry console, the paths are in the `app-engine-tmp` directory; i.e. GAE is using the 

[This answer at googlecloudcommunity.com](https://www.googlecloudcommunity.com/gc/Developer-Tools/Container-Registry-deprecation-usage-in-App-Engine-Deployment/m-p/771189/) says we shouldn't have to do anything...? I'm unclear whether that's official.

Another answer pointed [here](https://cloud.google.com/appengine/docs/standard/testing-and-deploying-your-app#manage_build_images), which I think tells us that the relevant usage is "of gcr.io domains". There's a [happy path](https://cloud.google.com/artifact-registry/docs/transition/auto-migrate-gcr-ar#migrate-gcrio-hosted-ar) for that.

**Comment by stillgreenmoss** [comment]
hmmmm all i remember is that when i investigated this a few months ago i concluded that we did need to change something about how pairing bot builds for this. i think that was based on this blog post https://cloud.google.com/artifact-registry/docs/transition/transition-from-gcr

**Comment by cceckman** [comment]
Yeah - sorry, I didn't update here:

I used the "transition to gcr.io" option [here](https://cloud.google.com/artifact-registry/docs/transition/transition-from-gcr#transition-options) for the testbot project. #56 made it out to testbot successfully -- I can see the changes in the logs.

I think that option winds up ~transparently pointing the relevant bucket to Artifact Registry instead of Container Registry? So the GAE configuration / Cloud Build configuration doesn't actually have to change, just the backend for the image store has changed.

I only ran it for the test project, not for the prod project; that's still TODO. (FWIW, I had to do a bunch of retries/approves/etc to get through the automated tool for the migration, but it did go through eventually.)
